src/eos-updater-apply.c

```diff
@@ -148,6 +152,19 @@ apply (GTask *task,
 
   newbootver = ostree_deployment_get_deployserial (new_deployment);
 
+  if (!eos_extensions_save (data->extensions,
+                            repo,
+                            cancel,
+                            &error))
+    goto error;
+  g_set_object (&data->branch_file, data->extensions->branch_file);
+  g_clear_object (&data->extensions);
```

Comment:
Is there any reason why we clear `data->extensions` after calling `eos_extensions_save` successfully?